constructor/winexe.py

```diff
@@ -330,32 +311,71 @@ def make_nsi(
-    ppd['register_python_default'] = info.get('register_python_default', None)
-    ppd['check_path_length'] = info.get('check_path_length', None)
-    ppd['check_path_spaces'] = info.get('check_path_spaces', True)
-    ppd['keep_pkgs'] = info.get('keep_pkgs') or False
-    ppd['pre_install_exists'] = bool(info.get('pre_install'))
-    ppd['post_install_exists'] = bool(info.get('post_install'))
-    ppd['with_conclusion_text'] = bool(conclusion_text)
-    ppd["enable_debugging"] = bool(os.environ.get("NSIS_USING_LOG_BUILD"))
-    ppd["has_conda"] = info["_has_conda"]
-    ppd["custom_welcome"] = info.get("welcome_file", "").endswith(".nsi")
-    ppd["custom_conclusion"] = info.get("conclusion_file", "").endswith(".nsi")
-    ppd["has_license"] = bool(info.get("license_file"))
-    ppd["post_install_pages"] = bool(info.get("post_install_pages"))
-
-    data = preprocess(data, ppd)
-    data = fill_template(data, replace, exceptions=nsis_predefines)
+        variables[key] = win_str_esc(value)
+
+    # From now on, the items added to variables will NOT be escaped
+
+    # These are mostly booleans we use with if-checks
+    variables.update(ns_platform(info['_platform']))
+    variables['initialize_conda'] = info.get('initialize_conda', True)
+    variables['initialize_by_default'] = info.get('initialize_by_default', None)
+    variables['register_python'] = info.get('register_python', True)
+    variables['register_python_default'] = info.get('register_python_default', None)
+    variables['check_path_length'] = info.get('check_path_length', None)
+    variables['check_path_spaces'] = info.get('check_path_spaces', True)
+    variables['keep_pkgs'] = info.get('keep_pkgs') or False
+    variables['pre_install_exists'] = bool(info.get('pre_install'))
+    variables['post_install_exists'] = bool(info.get('post_install'))
+    variables['with_conclusion_text'] = bool(conclusion_text)
+    variables["enable_debugging"] = bool(os.environ.get("NSIS_USING_LOG_BUILD"))
+    variables["has_conda"] = info["_has_conda"]
+    variables["custom_welcome"] = info.get("welcome_file", "").endswith(".nsi")
+    variables["custom_conclusion"] = info.get("conclusion_file", "").endswith(".nsi")
+    variables["has_license"] = bool(info.get("license_file"))
+    variables["post_install_pages"] = bool(info.get("post_install_pages"))
+
+    approx_pkgs_size_kb = approx_size_kb(info, "pkgs")
+
+    # UPPERCASE variables are unescaped (and unquoted)
+    variables['NAME'] = name
+    variables['NSIS_DIR'] = NSIS_DIR
+    variables['BITS'] = str(arch)
+    variables['PKG_COMMANDS'] = '\n    '.join(pkg_commands(download_dir, dists))
+    variables['SIGNTOOL_COMMAND'] = signing_tool.get_signing_command() if signing_tool else ""
+    variables['SETUP_ENVS'] = '\n    '.join(setup_envs_commands(info, dir_path))
+    variables['WRITE_CONDARC'] = '\n    '.join(add_condarc(info))
+    variables['SIZE'] = str(approx_pkgs_size_kb)
```

Comment:
Do we need that conversion or will jinja handle it for us?

Reply:
Probably not needed, true.
```suggestion
    variables['SIZE'] = approx_pkgs_size_kb
```